### ISSUE TITLE
next-image-loader does not respect assetPrefix

### ISSUE THREAD

**Issue by ScriptedAlchemy** [vercel/next.js#28305]
### What version of Next.js are you using?

11.1.0

### What version of Node.js are you using?

14

### What browser are you using?

Chrome

### What operating system are you using?

macOS

### How are you deploying your application?

Other (serverless aws) 

### Describe the Bug

When specifying an asset prefix, images do not respect it.

```
 assetPrefix: isProd ? '/static/web-app/' : '',
```

images still emit paths like `_next/` instead of `/static/web-app/_next` 

### Expected Behavior

Asset prefixes should be added to image paths. 

I've been able to fix it with the following PR https://github.com/vercel/next.js/pull/28306

### To Reproduce

Try adding an asset prefix via next config and check the image paths that are emitted.

**Comment by timneutkens** [comment]
This is expected behavior as `assetPrefix` is being used by many users as a path to a static CDN and `/_next/image` needs server processing. This is similar to `/_next/data` which also needs to hit the Next.js server.

**Comment by PsyGik** [comment]
So if one were to deploy multiple Next.js apps on a single domain, each app with it's own path...it isn't supported by Next.js?

```
App 1 - https://fancy.app/app-one
App 2 - https://fancy.app/app-two
App 3 - https://fancy.app/app-three
```

Based on how Next.js works today, `_next/data/*` for all the 3 apps would be `https://fancy.app/_next/data/*`, instead of respecting the path, `https://fancy.app/app-one/_next/data/*`.

So how is one supposed to deduce which request is for which app?

**Comment by styfle** [comment]
Fixed in #29307, thanks!

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
